tests/summmarizer/test_image_description_retreiver.py

```diff
@@ -0,0 +1,99 @@
+import json
+from unittest.mock import Mock, patch
+
+import pytest
+
+from classifier_and_summarizer.summarizer.web_entity_detection import (
+    BadRequestError, ImageDescriptionRetriever)
+
+
+def mocked_requests_post(*args, **kwargs):
+    json_data = json.loads(kwargs['data'])
+    label = ""
+    entity = ""
+    status_code = 200
+    if json_data["requests"][0]["image"]["source"]["imageUri"] \
+            == "https://tinyurl.com/y7how2rj":
+        label = "sundar pichai"
+        entity = "sundar pichai Alphabet"
+    elif json_data["requests"][0]["image"]["source"]["imageUri"] \
+            == "https://tinyurl.com/y9bvoehm":
+        label = "larry page"
+        entity = "larry page google"
+    elif json_data["requests"][0]["image"]["source"]["imageUri"] \
+            == "https://tinyurl.com/y9t35t3z":
+        label = "sergey brin"
+        entity = "sergey brin google"
+    else:
+        status_code = 400
+
+    response = json.dumps({
+        "responses": [
+            {
+                "webDetection": {
+                    "webEntities": [
+                        {
+                            "description": entity
+                        }
+                    ],
+                    "bestGuessLabels": [
+                        {
+                            "label": label,
+                            "languageCode": "en"
+                        }
+                    ]
+                }
+            }
+        ]
+    })
+    return Mock(status_code=status_code, content=response)
+
+
+@patch(
+    'classifier_and_summarizer.summarizer.web_entity_detection.requests.post',
+    side_effect=mocked_requests_post)
+def test_response_format(mocked_post):
```

Comment:
We don't need to verify the response from the server. I meant that we need to verify the request we send to the server.

Reply:
Apologies. Updated accordingly!